src/backend/optimizer/plan/createplan.c

```diff
@@ -640,6 +640,8 @@ use_physical_tlist(PlannerInfo *root, RelOptInfo *rel)
 static void
 disuse_physical_tlist(PlannerInfo *root, Plan *plan, Path *path)
 {
+	Insist(plan);
```

Comment:
How about putting the call under a general condition instead?
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/optimizer/plan/createplan.c#L3697-L3703
```c

	disuse_physical_tlist(root, inner_plan, best_path->jpath.innerjoinpath);
	if (outer_plan)
	{
		disuse_physical_tlist(root, outer_plan, best_path->jpath.outerjoinpath);

		/* If we expect batching, suppress excess columns in outer tuples too */
		if (best_path->num_batches > 1)
			disuse_physical_tlist(root, outer_plan, best_path->jpath.outerjoinpath);
	}
```

Reply:
Are you sure that at this point outer_plan can't be NULL?  I understand that it's a hashjoin node but I can't say for sure that it wasn't intended

Reply:
I think create_plan_recurse cannot return NULL

Reply:
Done